src/components/carousel/ThumbnailPagination.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {
+  FC,
+  LegacyRef,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { EmblaCarouselType } from 'embla-carousel-react';
+
+import { useMultiStyleConfig } from '@chakra-ui/react';
+
+import ThumbnailNavigationButton from './ThumbnailNavigationButton';
+import Thumbnail from './Thumbnail';
+import Flex from '../flex';
+import Box from '../box';
+
+interface Props {
+  currentSlideIndex: number;
+  thumbnails: ReactNode[];
+  mainCarousel?: EmblaCarouselType;
+  paginationCarouselRef?: LegacyRef<HTMLDivElement>;
+  paginationCarousel?: EmblaCarouselType;
+}
+
+const ThumbnailPagination: FC<Props> = ({
+  currentSlideIndex,
+  thumbnails,
+  mainCarousel,
+  paginationCarouselRef,
+  paginationCarousel,
+}) => {
+  const [thumbnailScrollProgress, setThumbnailScrollProgress] = useState(0);
+  const [showPaginationButtons, setShowPaginationButtons] = useState(true);
+  const { pagination } = useMultiStyleConfig('Carousel', {
+    variant: 'fullScreen',
+  });
+
+  const scrollPrev = useCallback(
+    () => paginationCarousel && paginationCarousel.scrollPrev(),
+    [paginationCarousel]
+  );
+  const scrollNext = useCallback(
+    () => paginationCarousel && paginationCarousel.scrollNext(),
+    [paginationCarousel]
+  );
+
+  const onThumbnailClick = useCallback(
+    (index: number) => {
+      if (!mainCarousel || !paginationCarousel) return;
+      if (paginationCarousel.clickAllowed()) {
+        mainCarousel.scrollTo(index);
+      }
+    },
+    [mainCarousel, paginationCarousel]
+  );
+
+  const onScroll = useCallback(() => {
+    if (
+      !paginationCarousel ||
+      paginationCarousel.slidesNotInView().length === 0
+    ) {
+      setShowPaginationButtons(false);
+      return;
+    }
+
+    const progress = Math.max(
+      0,
+      Math.min(1, paginationCarousel.scrollProgress())
+    );
+    setThumbnailScrollProgress(progress);
+  }, [paginationCarousel]);
+
+  useEffect(() => {
+    if (!paginationCarousel) return;
+    onScroll();
+    paginationCarousel.on('scroll', onScroll);
+  }, [paginationCarousel, onScroll]);
+
+  return (
+    <Box ref={paginationCarouselRef} __css={pagination} aria-label="Pagination">
+      <Flex alignItems="center" height="full" id="thumbnail-wrapper">
+        {thumbnails.map((slide, index) => (
+          <Thumbnail
+            key={`slide-${index}`}
+            onClick={() => onThumbnailClick(index)}
+            isCurrent={index === currentSlideIndex}
+            thumbnailIndex={index}
+            totalThumbnails={thumbnails.length}
+          >
+            {paginationCarouselRef ? (
+              slide
+            ) : (
+              <Box h="full" w="full" backgroundColor="gray.50" />
```

Comment:
Because `refs` are only defined on the client, I want to render empty boxes on SSR. With this, we can avoid the weird behavior that they pop up one by one and shift around. You can see that at CFY for example: https://www.carforyou.ch/de/auto/cabriolet/mercedes-benz/sl/63-v8-amg-cabriolet-3445619